subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/CacheAccessWorker.java

```diff
@@ -213,12 +213,12 @@ public synchronized void stop() {
             try {
                 workQueue.put(new ShutdownOperationsCommand());
             } catch (InterruptedException e) {
-                // ignore
+                Thread.currentThread().interrupt();
             }
             try {
                 doneSignal.await();
             } catch (InterruptedException e) {
-                // ignore
+                Thread.currentThread().interrupt();
```

Comment:
Wonder what unexpected behavior we'll now discover due to changes like these...

Reply:
This actually never gets interrupted in practice, we always let our cache worker threads finish.

Even if it did, restoring the Thread's interrupt status is the right thing to do. The only time one should swallow that status is at the very top of the chain. For instance like we do in our TimeoutTaskExecutor, which continues with the next task after a timeout has happened.

I did spend a lot of time going through each class and understanding its purpose before deciding on the interrupt behavior. Most notably AsyncDispatch had some very interesting requirements. Of course there may still be issues introduced by this, but if so they are not showing up in any test, which I take as a good sign :)